api/src/main/java/org/openmrs/validator/PersonValidator.java

```diff
@@ -107,7 +108,36 @@ public void validate(Object target, Errors errors) {
 		
 		ValidateUtil.validateFieldLengths(errors, Person.class, "gender", "personVoidReason");
 	}
-	
+
+	/**
+	 * Checks if the death date is in accordance with dead status.
+	 *
+	 * @param errors Stores information about errors encountered during validation
+	 * @param deathDate the deathdate to validate
+	 * @param dead is the dead status
+	 */
+	private void validateDeathDateDead(Errors errors, Date deathDate, boolean dead) {
+		if(deathDate == null || dead == true) {
```

Comment:
I asked because of your response which was: (the actual case we want to validate is dead == false and deathdate == null.)